src/Assets.ts

```diff
@@ -0,0 +1,234 @@
+    );
+
+    return entries.reduce<TickerReservation[]>((result, [key, relation]) => {
+      if (relation.isTickerOwned) {
+        const ticker = tickerToString(key.args[1]);
+
+        if (isPrintableAscii(ticker)) {
+          return [...result, new TickerReservation({ ticker }, context)];
+        }
+      }
+
+      return result;
+    }, []);
+  }
+
+  /**
+   * Retrieve a Ticker Reservation
+   *
+   * @param args.ticker - Security Token ticker
+   */
+  public async getTickerReservation(args: { ticker: string }): Promise<TickerReservation> {
+    const { ticker } = args;
+    const {
+      context: {
+        polymeshApi: {
+          query: { asset },
+        },
+      },
+      context,
+    } = this;
+
+    const { owner, expiry } = await asset.tickers(stringToTicker(ticker, context));
+
+    if (!owner.isEmpty) {
+      if (!expiry.isNone) {
+        return new TickerReservation({ ticker }, context);
+      }
+
+      throw new PolymeshError({
+        code: ErrorCode.UnmetPrerequisite,
+        message: `${ticker} token has been created`,
+      });
+    }
+
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: `There is no reservation for ${ticker} ticker`,
+    });
+  }
+
+  /**
+   * Retrieve all the Security Tokens owned by an Identity
+   *
+   * @param args.owner - identity representation or Identity ID as stored in the blockchain
+   *
+   * @note tokens with unreadable characters in their tickers will be left out
+   */
+  public async getSecurityTokens(args?: { owner: string | Identity }): Promise<SecurityToken[]> {
+    const {
+      context: {
+        polymeshApi: { query },
+      },
+      context,
+    } = this;
+
+    const did = await getDid(args?.owner, context);
+
+    const entries = await query.asset.assetOwnershipRelations.entries(
+      stringToIdentityId(did, context)
+    );
+
+    return entries.reduce<SecurityToken[]>((result, [key, relation]) => {
+      if (relation.isAssetOwned) {
+        const ticker = tickerToString(key.args[1]);
+
+        if (isPrintableAscii(ticker)) {
+          return [...result, new SecurityToken({ ticker }, context)];
+        }
+      }
+
+      return result;
+    }, []);
+  }
+
+  /**
+   * Retrieve a Security Token
+   *
+   * @param args.ticker - Security Token ticker
+   */
+  public async getSecurityToken(args: { ticker: string }): Promise<SecurityToken> {
+    const { ticker } = args;
+
+    const token = new SecurityToken({ ticker }, this.context);
+    const exists = await token.exists();
+
+    if (!exists) {
+      throw new PolymeshError({
+        code: ErrorCode.DataUnavailable,
+        message: `There is no Security Token with ticker "${ticker}"`,
+      });
+    }
+
+    return token;
+  }
+
+  /**
+   * Create a Security Token
+   *
+   * @note if ticker is already reserved, then required role:
+   *   - Ticker Owner
+   */
+  public createToken: ProcedureMethod<CreateSecurityTokenWithTickerParams, SecurityToken>;
```

Comment:
Since we are refactoring stuff, I think we should move this up to line 70 so that all such funcs are at same place (just a suggestion)

Reply:
Makes sense to me.